demo/all_demos/templates/index.html

```diff
@@ -0,0 +1,93 @@
+<!DOCTYPE html>
+<html>
+<head>
+    {%if is_space %}
+        <meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests">
+    {% endif %}
+    <style>
+        /* Reset some default styles */
+        body, h1, h2, h3, p, ul {
+            margin: 0;
+            padding: 0;
+        }
+
+        body {
+            font-family: 'Arial', sans-serif;
+            overflow-y: hidden;
+        }
+
+        /* Styling for the sidebar */
+        .sidebar {
+            position: fixed;
+            top: 0;
+            left: 0;
+            height: 100vh;
+            width: 300px;
+            overflow-y: auto;
+            overflow-x: hidden;
+            font-size: 0.875rem; /* 14px */
+            line-height: 1.25rem; /* 20px */
+        }
+
+        .sidebar a {
+            display: block;
+            padding: 8px 16px;
+            text-decoration: none;
+            color: rgb(107 114 128);
+        }
+
+        .sidebar a:hover {
+            color: black;
+            transform: translateX(1px);
+            
+        }
+
+        /* Apply a different style to the selected item in the sidebar */
+        .sidebar a.active {
+            background-color: rgb(251 146 60);
+            color: white;
+            border-radius: 0.75rem;
+            font-weight: bold;
+        }
+
+        /* Styling for the content */
+        .content {
+            margin-left: 300px;
+            padding: 20px;
+            display: block;
+            height: 100vh;
+        }
+
+        /* Make the iframe responsive */
+        .content iframe {
+            width: 100%;
+            border: 0;
+            height: 100%;
+        }
+
+        @media only screen and (max-width: 600px) {
+            /* Adjust styles for smaller screens */
+            .sidebar {
+                width: 100%;
+                position: relative;
+                height: auto;
+            }
+
+            .content {
+                margin-left: 0;
+            }
+        }
+    </style>
+    <script src="//unpkg.com/alpinejs" defer></script>
+</head>
+<body x-data="{current_demo: 'altair_plot' }">
+    <div class="sidebar">
+        {% for name in names %}
+            <a @click="current_demo = '{{ name }}'" :class="current_demo == '{{ name }}' ? 'active' : ''">{{ name }}</a>
+        {% endfor %}
+    </div>
+    <div class="content">
+        <iframe :src="`/demo/${current_demo}?__theme=light`"></iframe>
```

Comment:
The color theme is hardcoded as `light` here.
Instead passing `document.location.search` would be good?

Also, perhaps passing both the query string and the hash same as the parent page would be better as Spaces may set them as some special values.

Reply:
Ok I used `document.location.search`. If the dark mode theme is used, the side bar and page background won't match but I think it's ok for now. I think a dark mode theme for the sidebar is overkill for now.

Reply:
nit: What about adding `location.hash`? HF Spaces potentially set it, though Gradio currently doesn't have code dealing with it?

Reply:
Why not use the webcomponent? It will handle the theme and stuff by default? Iframe prevents it from doing that, spaces has a whole bunch of logic to propagate changes on the hosting page down to the embedded iframe that we don't want to reimplement here.

Reply:
The webcomponent was not working in this case. Simple repro here (https://github.com/gradio-app/gradio/issues/6688). Also we don't publish the cdn assets on every commit yet so it's possible it breaks due to some change in the PR that hasn't been deployed yet?

Agreed using the web component is the right thing in the future.

Reply:
> nit: What about adding location.hash as well?

Yea we don't use this yet so doesn't feel necessary.

Reply:
> Also we don't publish the cdn assets on every commit yet so it's possible it breaks due to some change in the PR that hasn't been deployed yet?

We wouldn't use the published version, we could just use the local version store in the files.

Using the iframe is problematic because of the logic I mentioned earlier. There is a lot of orchestration between the host and embedded page that hasn't been implemented in this PR.

Reply:
Yes I think it would be easier if you can get the webcomponent to work! I was not able to because of the issues mentioned in #6688

Reply:
Will sort that!

Reply:
Thank you @pngwn !